metrics/ci_integrations/lib/jenkins/adapter/jenkins_ci_client_adapter.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:ci_integration/common/client/ci_client.dart';
+import 'package:ci_integration/common/model/interaction_result.dart';
+import 'package:ci_integration/coverage_json_summary/model/coverage_json_summary.dart';
+import 'package:ci_integration/jenkins/client/jenkins_client.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_build.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_build_result.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_building_job.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_query_limits.dart';
+import 'package:metrics_core/metrics_core.dart';
+
+/// An adapter for [JenkinsClient] to fit [CiClient] contract.
+class JenkinsCiClientAdapter implements CiClient {
+  /// A Jenkins client instance used to perform API calls.
+  final JenkinsClient jenkinsClient;
+
+  /// Creates an instance of this adapter with the given [jenkinsClient].
+  ///
+  /// Throws [ArgumentError] if the given Jenkins client is `null`.
+  JenkinsCiClientAdapter(this.jenkinsClient) {
+    ArgumentError.checkNotNull(jenkinsClient, 'jenkinsClient');
+  }
+
+  @override
+  Future<List<BuildData>> fetchBuildsAfter(
+    String projectId,
+    BuildData build,
+  ) async {
+    ArgumentError.checkNotNull(build, 'build');
+
+    final buildingJob = await _fetchBuilds(
+      projectId,
+      limits: JenkinsQueryLimits.endBefore(0),
+    );
+    final lastBuild = buildingJob.lastBuild;
+    int numberOfBuilds = lastBuild.number - build.buildNumber;
+
+    if (numberOfBuilds <= 0) return [];
+
+    List<JenkinsBuild> builds;
+    while (builds == null) {
+      final buildingJob = await _fetchBuilds(
```

Comment:
```suggestion
      final buildingJobForNumberOfBuilds = await _fetchBuilds(
```
How about changing it like this, as we already have buildingJob defined earlier in this method so it might make sense to change the name or we can `Extract Method` here to make this while more granular?